parser/src/parser.rs

```diff
@@ -1312,8 +1308,12 @@ def foo():
 
 # Indented magic
 for a in range(5):
-    %ls
-    pass
+    !ls
+
+p1 = !pwd
+p2: str = !pwd
```

Comment:
If there's any comment after the statement, they'll be consumed and be part of the magic command as is done by the original implementation:

```python
t.transform_cell("f: str = %foo  # comment")
# "f: str = get_ipython().run_line_magic('foo', ' # comment')\n"
```

Reply:
Do you want to add a test for this case?